### ISSUE TITLE
Alternator: Inappropriate error message for UpdateTable on non-existent table

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#9747]
When `UpdateTable` is used on a non-existent table, the result should be a `ResourceNotFoundException`, but today we get the ugly:

```
Internal server error: std::_Nested_exception<no_such_column_family> (Can't find a column family 97EDHOYGHRLQEN51X27B in keyspace alternator_97EDHOYGHRLQEN51X27B): std::out_of_range ()
```

The problem is that `executor::update_table()` calls `db.find_column_family(keyspace_name, table_name)`, which throws an exception instead of an Alternator API error, so this is translated into an "internal server error". Instead, we should use `executor::find_table()` or if we can't use it directly, copy the code generating the `api_error::resource_not_found`.

**Comment by bhalevy** [comment]
I'm not sure why the nested exception were used back in d3fe0c5182966d66adcf0bc097264efd545c4f2a.
@elcallio do you remember why?
I hope we can get rid of them.

**Comment by nyh** [comment]
I agree the nested exception doesn't make sense in this case because the code has
```c++
const keyspace& database::find_keyspace(const sstring& name) const throw (no_such_keyspace) {
    try {
        return _keyspaces.at(name);
    } catch (...) {
        std::throw_with_nested(no_such_keyspace(name));
    }
}
```

And the only reason at() can fail (I think...) is that the keyspace is missing.

In any case, as I noted elsewhere, the problem is not just the ugly text of the message, but the exception type itself. In Alternator (this issue is about the Alternator case) it is important to throw exceptions of type "api_error" to send the user a real according-to-the-protocol failure. Any other exception is presented to the user as an "internal server error", which isn't the end of the world, not not usually what we want to happen.

**Comment by bhalevy** [comment]
> In any case, as I noted elsewhere, the problem is not just the ugly text of the message, but the exception type itself.

Agreed.  I wonder where this happens and why the different exceptions end up with different http error codes.

**Comment by bhalevy** [comment]
Answering myself: validate_keyspace throws `bad_param_exception` that is derived from `bad_request_exception`, resulting in `reply::status_type::bad_request`

**Comment by nyh** [comment]
> Answering myself: validate_keyspace throws `bad_param_exception` that is derived from `bad_request_exception`, resulting in `reply::status_type::bad_request`

So it's exactly like in Alternator - REST API requests should fail with a bad_request_exception (and possibly other types of supported exceptions?) and failing with any other exception is wrong and results in error code 500 (internal server error).

**Comment by bhalevy** [comment]
See https://github.com/scylladb/scylla/pull/9746/commits/850acac32ea8c68705bcba6e15dce8749afbb475

**Comment by nyh** [comment]
Backported to the not-yet-released next-4.6 (commit 31eeb44d28c4c83f01f0e33417b34ae4351459b3)
Not backporting to earlier branches because this is just an error message cleanup.

**Comment by nyh** [comment]
Again, this is an old issue from which I already removed the "Backport candidate" tag last year after backporting, but some tagging accident (?) in July re-added the tag. I'll remove it again.
